4 - AnimalShelter/Code/AnimalShelter.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+//rework enums
+//restore proper list
+
+namespace AnimalShelter.Code
+{
+    public class AnimalShelter
+    {
+        //private List<Animal> _animals = new List<Animal>();
+        private List<Animal> _animals = new List<Animal>();
+        public List<Animal> animals
+        {
+            get { return _animals; }
+            set { _animals = value; }
+        }
+        public AnimalShelter()
+        {
+               
+        }
+        public Result AddAnimal(Animal animal)
+        {
+            if (animal.AnimalType == AnimalType.Deer || animal.AnimalType == AnimalType.Bear)
+            {
+                return new Result( "Animal was not a supported animal");
+            }
+            else
+            {
+                _animals.Add(animal);//using mine
+                return new Result(animal);
+            }
+        }
```

Comment:
Super clean code and easy to maintain. WITH THAT SAID, I would have reversed the logic. Imagine a new animal type is added to the enum, due to your logic it is now automatically supported - and this is mostly likely not something that would want to be intended.